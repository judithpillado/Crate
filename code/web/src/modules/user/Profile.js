// Imports
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'
import { Link } from 'react-router-dom'

// UI Imports
import { Grid, GridCell } from '../../ui/grid'
import { H3, H4 } from '../../ui/typography'
import Button from '../../ui/button'
import { grey, grey2 } from '../../ui/common/colors'

// App Imports
import userRoutes from '../../setup/routes/user'
import { logout } from './api/actions'

// Component
// need to create edit profile button that links to edit profile form 
// need to create deliveries button 
// need to create histories button 
// possibly do conditional rendering in header. If user is logged in, they can see their history and deliveries via the header/nav bar

// helmet - do we put the profile picture here or in the top title bar?
const Profile = (props) => (
  <div>
    {/* SEO */}
    <Helmet>
      <title>My Profile - Crate</title>
    </Helmet>

    {/* Top title bar */}
    <Grid style={{ backgroundColor: grey }}>
      <GridCell style={{ padding: '2em', textAlign: 'center' }}>
        <H3 font="secondary">My profile</H3>
      </GridCell>
    </Grid>

    <Grid>
      <GridCell style={{ padding: '2em', textAlign: 'center' }}>
        <H4 style={{ marginBottom: '0.5em' }}>{props.user.details.name}</H4>

        <p style={{ color: grey2, marginBottom: '2em' }}>{props.user.details.email}</p>

        <Link to={userRoutes.subscriptions.path}>
          <Button theme="primary">Subscriptions</Button>
        </Link>

        <Button theme="secondary" onClick={props.logout} style={{ marginLeft: '1em' }}>Logout</Button>
      </GridCell>
    </Grid>
  </div>
)

// Component Properties
Profile.propTypes = {
  user: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
}

// Component State
// looks like it's accessing state from the store and holding it here
function profileState(state) {
  return {
    user: state.user
  }
}

// might need to create form file and import into profile 

export default connect(profileState, { logout })(Profile) 