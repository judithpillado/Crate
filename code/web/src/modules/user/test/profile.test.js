import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import Profile from '../profile.js'
import { profileState } from '../profile.js'
import { BrowserRouter } from 'react-router-dom'
import { Router } from 'react-router-dom'
import { createStore } from 'redux'
import rootReducer from '../../../setup/store.js'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

const mockStore = configureStore([])

describe("Profile", () => {
  let store;
  let component;
  it('should display profile text and buttons', () => {
    const rootReducer = jest.fn()
    const store = mockStore({
      user: {
        details: {
          name: 'Joe User',
          email: 'Joe@email.com',
          role: 'USER'
        }
      }
    })

    component = renderer.create(
      <Provider store={store} >
        <BrowserRouter>
            <Profile/>
        </BrowserRouter>
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
//
