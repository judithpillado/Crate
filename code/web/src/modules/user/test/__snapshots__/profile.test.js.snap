// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should display profile text and buttons 1`] = `
<div>
  <div
    className="jsx-715511798 "
    style={
      Object {
        "backgroundColor": "#f0f0f0",
      }
    }
  >
    <div
      className="jsx-230203545 "
      style={
        Object {
          "padding": "2em",
          "textAlign": "center",
        }
      }
    >
      <h3
        className="jsx-1368067487 "
      >
        My profile
      </h3>
    </div>
  </div>
  <div
    className="jsx-715511798 "
  >
    <div
      className="jsx-230203545 "
      style={
        Object {
          "padding": "2em",
          "textAlign": "center",
        }
      }
    >
      <h4
        className="jsx-3486686285 "
        style={
          Object {
            "marginBottom": "0.5em",
          }
        }
      >
        Joe User
      </h4>
      <p
        style={
          Object {
            "color": "#999999",
            "marginBottom": "2em",
          }
        }
      >
        Joe@email.com
      </p>
      <a
        href="/user/subscriptions"
        onClick={[Function]}
      >
        <button
          className="jsx-1228826222 "
          disabled={false}
          type="button"
        >
          Subscriptions
        </button>
      </a>
      <button
        className="jsx-2258186958 "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "1em",
          }
        }
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
